### PR TITLE
Fix return of `BaseHtml::getAttributeValue`

### PR DESCRIPTION
**Pull request by WinterSilence**
Similar to freezed #18831 + test fix + update changelog

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #18831

### CODE REVIEW

File: tests/framework/helpers/HtmlTest.php [hunk 1]
```diff
@@ -1887,15 +1887,15 @@ public function testGetAttributeValue()
...
+        $expected = '1';
         $actual = Html::getAttributeValue($model, 'types');
         $this->assertSame($expected, $actual);
 
         $model->types = [
             $activeRecord,
         ];
 
-        $expected = [1];
+        $expected = '[1]';
```
Comment (rob006) [inline tests/framework/helpers/HtmlTest.php:1898]: Why did you changed this?

Comment (WinterSilence) [reply tests/framework/helpers/HtmlTest.php:1898]: @rob006 
изображение

Comment (WinterSilence) [reply tests/framework/helpers/HtmlTest.php:1898]: @rob006 see #18831 for more info

Comment (rob006) [reply tests/framework/helpers/HtmlTest.php:1898]: @WinterSilence That does not explain anything. Method had documented array as return type, and returned array (which is confirmed by tests). Now you changed both phpdoc and actual return type, without explaining why it should be changed?

Comment (WinterSilence) [reply tests/framework/helpers/HtmlTest.php:1898]: @rob006 how html attribute can be array?

Comment (WinterSilence) [reply tests/framework/helpers/HtmlTest.php:1898]: @rob006

> Well, it was passing before so you change the situation and you must update tests to reflect it (and most probably add new ones). Please do so.

this pr just do it

Comment (rob006) [reply tests/framework/helpers/HtmlTest.php:1898]: > how html attribute can be array?

For example `activeCheckboxList()` uses `getAttributeValue()` internally and expects array (you can have multiple options selected).

On the other hand, how on earth `'[1]'` could be sensible response if value is array?

Comment (rob006) [reply tests/framework/helpers/HtmlTest.php:1898]: Also, I don't think that changing type from int to string in previous tests is correct either. It just make it less specific for no good reason.

Comment (WinterSilence) [reply tests/framework/helpers/HtmlTest.php:1898]: @rob006 attribute value is always string, you cant use something like this `data-foo[0]="1" data-foo[1]="1"` or `data-foo=["1" , "2"]`

Comment (WinterSilence) [reply tests/framework/helpers/HtmlTest.php:1898]: @bizley come here, after 
> Well, it was passing before so you change the situation and you must update tests to reflect it (and most probably add new ones). Please do so.

but here you go other side? wtf?

Comment (WinterSilence) [reply tests/framework/helpers/HtmlTest.php:1898]: @rob006 dislikes? really? you're not in school anymore, come back and let's talk

Comment (rob006) [reply tests/framework/helpers/HtmlTest.php:1898]: @WinterSilence I don't want to waste more time on discussions with you. I gave you an example what case will be broken by this change and why `getAttributeValue()` should return array in some scenarios. If you didn't bother to test it or even understand it, then I don't think there is anything more to discuss here.

**Comment by bizley** [comment]
@WinterSilence you are violating Yii code of conduct again and this time is the last one. From now on every issue of yours where you cannot behave and discuss things in a civil way will be closed, every insulting comment will be deleted, and you will be reported to Github.

**Comment by WinterSilence** [comment]
@rob006 @bizley why are you satisfied with #18831, but my version with fixed test does not? I think I's bulling and report to Github too.

**Comment by samdark** [comment]
https://github.com/yiisoft/yii2/issues/18829

**Comment by My6UoT9** [comment]
@WinterSilence 
http://catb.org/~esr/faqs/smart-questions.html#not_losing

**Pull request closed without merging**